### ISSUE TITLE
[Feat] Generic Way to trigger a light bulb for a code action on PyLint issues

### ISSUE THREAD

**Issue by dciborow** [microsoft/vscode-pylint#43]
I would like a generic way to specify per linter per error code to trigger a light bulb for a code action that is provided in a configuration file. Before finding a truely global way to achieve, this since PyLint handles the majority of my Python static code analysis, I would like to begin here. A subset of PyLint's suggestions could contain a suggested fix. In easy cases, this can be done by triggering a preexisting function in VSCode.

Here are some example PyLint Warnings.
<html>
<body>

Code | VS Action | Issue | Description
-- | -- | -- | --
C0301 | Format Document | line-too-long | *Line too long (%s/%s)*
C0303 | Format Document | trailing-whitespace | *Trailing whitespace*
C0304 | Format Document | missing-final-newline | *Final newline missing*
C0305 | Format Document | trailing-newlines | *Trailing newlines*
C0321 | Format Document | multiple-statements | *More than one statement on a single line*
C0410 | Sort Imports | multiple-imports | *Multiple imports on one line (%s)*
C0411 | Sort Imports | wrong-import-order | *%s should be placed before %s*
C0412 | Sort Imports | ungrouped-imports | *Imports from package %s are not grouped*
C0413 | Sort Imports | wrong-import-position | *Import "%s" should be placed at the top of the module*

</body>
</html>

Here is an example of some slightly more complex Fixes and Actions.
<html>
<body>

Code | VS Action | Issue | Description
-- | -- | -- | --
W0611 | Delete Line | unused-import | *Unused %s*
W0612 | Delete Line | unused-variable | *Unused variable %r*
W0613 | Refactor Function Signature - Remove %r | unused-argument | The argument in the function definition is deleted, then, anywhere in the code base that the argument was present, it is also removed
W0621 | Refactor Variable Name - %r to _%r | redefined-outer-name | This is a simple opinionated auto-fix suggestion, that should be further configurable by the user
W0622 | Refactor Variable Name - %r to my_%r | redefined-builtin | This is a simple opinionated auto-fix suggestion, that should be further configurable by the user

</body>
</html>

The HOOK into PyLint, would read the configuration file containing the list of PyLint Codes, and descriptions of the actions in VS Code. 

In the above examples the following actions would be used.
1. Format Document
2. Sort Imports
3. Delete Line
4. Refactor Function Signature
5. Refactor Variable Name

Some may use a sed like REGEX to perform a transformation, use case camelCase -> snake_case. 
1. Refactor Variable Name /[.*][^][.*]/(0)_(1)(2)/

If there is interest in developing the hook, I will continue to generate maps between PyLint Codes and VS actions. But I leave no suggested way to format the file, or hook declaration, and look for advice for that from the project owners.

**Comment by brettcannon** [comment]
And even if we don't decide to have some generic way for all linters, we should probably discuss whether we want to accept changes to this extension directly to provide quick fixes.

**Comment by dciborow** [comment]
One option to consider would be supporting sarif.

Or, even just the more generic "GitHub" logging sytle:
`%f:%l:%c: %m`

%f = filename
%l = line number
%c = column number
%m = message

And within the message, a suggestion can be placed between ``` like this

```
```suggestion
RUN conda install -y python=3.8 \
` ``
```
(space added to  ` `` to display properly)

**Comment by brettcannon** [comment]
Standardized output formats isn't going to be what helps here. The critical piece would be some mechanism to make it so you can hook into the linter results that **we** provide to get the quick fixes. Basically quick fixes are given the diagnostics, so you're already passed caring about _how_ we got the diagnostic.

**Comment by ketozhang** [comment]
I'm suggesting an additional action: to ignore the pylint rule for that line by adding the inline comment `# pylint: disable-<issue>`.

**Comment by ketozhang** [comment]
Additionally a second code action option for `#pylint: disable-next=<issue>`. The use case often being adding a comment will cause `line-too-long`.

Thoughts? I can work on this. @karthiknadig @luabud

**Comment by gilmarGNJ** [comment]
Any news on this?

**Comment by karthiknadig** [comment]
@gilmarGNJ The extension now supports code actions. The first set of code actions (once based on existing commands is already done. The one that needed text change, can also be implemented. This is unblocked for community contributions.

**Comment by dciborow** [comment]
I'm able to create #300

**Comment by karthiknadig** [comment]
Verification:

1. Install the latest pylint extension.
1. create a python file with following content:
```python
class MyObject(object):
    """Test class"""
    pass
```
1. In the problems window it should show: `useless-object-inheritance` 
1. Click on the quickfix provided for that problem.
1. Code should be updated to:
```python
class MyObject:
    """Test class"""
    pass
```
